code/API_definitions/location-subscriptions.yaml

```diff
@@ -0,0 +1,591 @@
+    
+    Webhook:
+      type: object
+      description: Detail for event channel
+      properties:
+        notificationUrl:
+          $ref: '#/components/schemas/NotificationUrl'
+        notificationAuthToken:
+          $ref: '#/components/schemas/NotificationAuthToken'
+      required: 
+        - notificationUrl
+    
+    EventSubscriptionId:
+      type: string
+      format: uuid
+      description: EventSubscription ID in UUID format      
+    
+    SubscriptionDetail:
+      type: object
+      description: Subscription detail
+      properties:
+        device:
+          $ref: '#/components/schemas/Device'
+        area:
+          $ref: '#/components/schemas/Area'
+        eventType: 
+          $ref: "#/components/schemas/EventType"
+      required:
+        - device
+        - area
+        - eventType
+    
+    EventType:
+      description: |
+        AREA_ENTERED - Location event triggered when the device entered the given area
+        AREA_LEFT - Location event triggered when the device left the given area
+      type: string
+      enum:
+        - AREA_ENTERED
+        - AREA_LEFT
+    
+    NotificationUrl:
+      type: string
+      example: "https://application-server.com"
+      description: https callback address where the event notification must be POST-ed
+    
+    NotificationAuthToken:
+      type: string
+      example: "c8974e592c2fa383d4a3960714"
+      description: Authentication token for callback API 
+    
+    CreateEventSubscription:
+      description: The request for creating a Location Event Subscription
+      type: object
+      properties:
+        webhook:
+          $ref: '#/components/schemas/Webhook'
+        subscriptionDetail:
+          $ref: '#/components/schemas/SubscriptionDetail'
+        subscriptionExpireTime:
+          type: string
+          format: date-time
+          example: 2023-01-17T13:18:23.682Z
+          description: The time when the location-tracking has to be terminated (e.g. in 14 days from now on). Provided in date-time format.
+      required:
+        - webhook
+        
+    SubscriptionInfo:
+      description: Represents a area-location event subscription.
+      type: object
+      allOf:
+        - $ref: '#/components/schemas/CreateEventSubscription'
+        - type: object
+          properties:
+            eventSubscriptionId:
+              $ref: '#/components/schemas/EventSubscriptionId'
+            startsAt:
+              type: string
+              format: date-time
+              example: 2023-01-17T13:18:23.682Z
+              description: The time when the location-tracking has to be started (e.g. in 7 days from now on). Provided in date-time format.
+            expiresAt:
+              type: string
+              format: date-time
+              example: 2023-01-17T13:18:23.682Z
+              description: The time when the location-tracking has to be expired (e.g. in 7 days from now on). Provided in date-time format.
```

Comment:
So it is always a date in future ? We assume there that a subscription terminated is de facto deleted. 
I'm fine with that but just to be sure we're on the same page.

Reply:
I changed like the comment above.